src/main/java/uk/gov/cshr/civilservant/controller/QuizController.java

```diff
@@ -40,12 +56,49 @@
 
     @PostMapping
     @Transactional
-    public ResponseEntity createQuiz(@Valid @RequestBody Quiz quiz) {
-        if (quiz.getProfession() != null && quiz.getProfession().getId() != null) {
-            quizRepository.deleteAllByProfessionId(quiz.getProfession().getId());
+    public ResponseEntity create(@Valid @RequestBody QuizDto quiz) {
+        quiz.setCreatedOn(LocalDateTime.now());
+
+        try {
+            Quiz quizRecord = quizDTOFactory.mapDtoToModel(quiz);
+            return ResponseEntity.ok(quizService.save(quizRecord).getId());
+        } catch (Exception ex) {
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body("There was a problem while creating the quiz." +
+                            "Check the values provided or please try again later");
         }
+    }
+
+    @GetMapping("/all")
+    public ResponseEntity<List<QuizDto>> getAllQuizzesInTheSystem() {
+        return ResponseEntity.ok(
+                quizService.getAll());
+    }
 
-        quizRepository.save(quiz);
-        return ResponseEntity.noContent().build();
+    @GetMapping("/profession/{professionId}")
+    public ResponseEntity getQuizForProfession(@PathVariable Long professionId) {
+        Optional<QuizDto> quiz = quizService.getQuizByProfessionId(professionId);
+        if (quiz.isPresent()) {
+            return ResponseEntity.ok(quizService.getQuizByProfessionId(professionId));
+        }
+        return ResponseEntity.notFound().build();
+    }
+
+    @Transactional
+    @DeleteMapping("/{id}/delete")
+    public ResponseEntity deleteQuiz(@PathVariable Long id) {
+        try {
+            log.info("Deleting Quiz with identifier",id);
```

Comment:
Could possibly change the DTO object to have an active/inactive boolean, on by default and switched off in deleteQuiz, when getting quiz or updating, always ensure that active is true.

Reply:
Yes. This is one of the design questions we went through with the customer yesterday. When we delete a quiz we mark it inactive with a flag and all the questions for the quiz will be marked as inactive as well. Will be handled as a part of the deleting user story

Reply:
The questions and quizzes will be needed for historical MI reporting.

Reply:
Awesome stuff